Controller/Adminhtml/Integration/Save.php

```diff
@@ -10,15 +11,25 @@
  *
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  */
-class Save extends IntegrationSave
-{
+class Save extends IntegrationSave {
     /**
-     * We ignore to validate the current user password
+     * If the integration is ours, ignore the current user password validation
      *
-     * @return bool
+     * @return $this
      */
-    protected function validateUser(): bool
-    {
-        return true;
+    protected function validateUser() {
+        $integration_name = $this->getRequest()->getParam("name");
+        $is_doofinder_integration = $this->getRequest()->getParam("is_doofinder_integration") === "true" ? true : false;
+        if ($is_doofinder_integration && $integration_name === "Doofinder Integration") {
+            return $this;
+        }
+
+        $password = $this->getRequest()->getParam(
+            \Magento\Integration\Block\Adminhtml\Integration\Edit\Tab\Info::DATA_CONSUMER_PASSWORD
+        );
+        $user = $this->_auth->getUser();
+        $user->performIdentityCheck($password);
+
+        return $this;
```

Comment:
```suggestion
        return parent::validateUser();
```
Creo que es mejor llamar la función padre para mantener la compatibilidad con versiones posteriores de Magento.